File: PLATFORM/프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int goal_dep;
    static int goal_target;
    static int ans;
   
    void dfs(int[] numbers, int idx, int dep, int sum){
        if(dep==goal_dep){
            if(sum==goal_target){
                ans++;
            }
            return;
            
        }
        
        dfs(numbers, idx+1, dep+1, sum+numbers[idx]);
        dfs(numbers, idx+1, dep+1, sum-numbers[idx]);
            
        
    }
   
    public int solution(int[] numbers, int target) {
        goal_dep = numbers.length;
        goal_target = target;
        dfs(numbers, 0, 0, 0);
        return ans;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py

cnt=0



def dfs(dep, idx, sum, numbers, target):
    global cnt
    
    if dep ==len(numbers) :
        if sum == target: 
            cnt+=1
        return
    
    dfs(dep+1, idx+1, sum+numbers[idx], numbers, target)
    dfs(dep+1, idx+1, sum-numbers[idx], numbers, target)
        
    

def solution(numbers, target):
    dfs(0,0,0, numbers, target)
    return cnt 